Bottom: 0688ce6af6e95d79135ff96c9e39fcc3c31084a8
Top:    3e90b33543a7ad7f7e207d2b8560406cce1239f4
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-23 17:56:35 +0100

draw.c: Fix order of farside object drawing

Backside silk should be drawn before bakside pads


---

diff --git a/src/draw.c b/src/draw.c
index 9f4a591..7380beb 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -365,13 +365,13 @@ DrawEverything (BoxTypePtr drawn_area)
       && gui->set_layer ("invisible", SL (INVISIBLE, 0), 0))
     {
       side = SWAP_IDENT ? COMPONENT_LAYER : SOLDER_LAYER;
-      r_search (PCB->Data->pad_tree, drawn_area, NULL, pad_callback, &side);
       if (PCB->ElementOn)
 	{
 	  r_search (PCB->Data->element_tree, drawn_area, NULL, element_callback, &side);
 	  r_search (PCB->Data->name_tree[NAME_INDEX (PCB)], drawn_area, NULL, name_callback, &side);
 	  DrawLayer (&(PCB->Data->Layer[max_copper_layer + side]), drawn_area);
 	}
+      r_search (PCB->Data->pad_tree, drawn_area, NULL, pad_callback, &side);
     }
 
   /* draw all layers in layerstack order */
